src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -233,12 +270,16 @@ RegisterShardPlacementXactCallback(void)
 void
 CompleteShardPlacementTransactions(XactEvent event, void *arg)
 {
-	if (shardPlacementConnectionList == NIL)
+	List *connectionList = NIL;
+
+	if (shardConnectionHash == NULL)
 	{
 		/* nothing to do */
 		return;
 	}
-	else if (event == XACT_EVENT_PRE_COMMIT)
+
+	connectionList = ConnectionList(shardConnectionHash);
+	if (event == XACT_EVENT_PRE_COMMIT)
```

Comment:
Made `ConnectionList` capable of receiving `NULL` so I could immediately assign up top, which let me remove this line entirely.